packages/core/useConfirmDialog/index.ts

```diff
@@ -0,0 +1,85 @@
+import { Ref } from 'vue-demi'
+import { createEventHook, EventHook } from '@vueuse/core'
+
+export interface useConfirmDialogReturn<T, D, E> {
+  /*
+  * Opens the dialog. Create promise and return it. Triggers `onReveal` hook.
+  */
+  reveal: (data?: T) => Promise<{ data: D | E | undefined; isCanceled: boolean }>
```

Comment:
Maybe extract `{ data: D | E | undefined; isCanceled: boolean }` to an interface and reuse it

Reply:
Good point!